QSV scale filter does not convert from full to limited range

### Problem Description

I have an issue with colors when converting on one machine, and no issue with other machine.

One machine (num-1) is windows laptop with handbrake 1.4.0
Another (num-2) is docker on linux with handbrake tag v1.24.0 ​(it reports 1.4.0 in GUI)
And third (num-3) is Ubuntu 20.04 with handbrake 1.3.1 from repos

When I convert mp4 video on num-1, I get the same color, as in source video.
When I convert the same mp4 video with the same settings in GUI and in CLI on num-2 and num-3, I get a brighter video, then the source. I check the same frame on the same display.

The result on any machine does not depend on encoder - same on h264, h265, h26x QSV, MPEG-4.

The only clue I have found is in the logs:
- On the num-1 the video section looks like this:
``` 
"Video": {
    "Encoder": "qsv_h264",
    "Level": "auto",
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "",
    "Preset": "quality",
    "Profile": "auto",
    "Quality": 20,
    "QSV": {
      "Decode": true
    }
```

- On the num-2 and num-3 the same section looks similar like this:
```
    "Video": {
        "ColorFormat": 0,
        "ColorMatrix": 1,
        "ColorPrimaries": 1,
        "ColorRange": 2,
        "ColorTransfer": 1,
        "Encoder": "qsv_h264",
        "Level": "auto",
        "Options": "",
        "Preset": "quality",
        "Profile": "auto",
        "QSV": {
            "AdapterIndex": 0,
            "AsyncDepth": 0,
            "Decode": false
        },
        "Quality": 20.0,
        "Tune": "",
        "Turbo": false,
        "TwoPass": false
    }
```

Why are those Color* settings present on linux machines? Is it the cause?
Can I some-how change it?

### Activity Log, Crash Log or any other details

Activity log on linux machine:

```
Handbrake Version: 1.4.0 (2021072000)
[21:01:26] gtkgui: Modified Custom Preset: /Custom/GoPro Custom
[21:01:26] Starting work at: Sun Jul 25 21:01:26 2021

[21:01:26] 1 job(s) to process
[21:01:26] json job:
{
    "Audio": {
        "AudioList": [
            {
                "Bitrate": 160,
                "CompressionLevel": -1.0,
                "DRC": 0.0,
                "DitherMethod": "auto",
                "Encoder": "fdk_aac",
                "Gain": 0.0,
                "Mixdown": "stereo",
                "NormalizeMixLevel": false,
                "PresetEncoder": "fdk_aac",
                "Quality": -3.0,
                "Samplerate": 0,
                "Track": 0
            }
        ],
        "CopyMask": [
            "copy:aac"
        ],
        "FallbackEncoder": "fdk_aac"
    },
    "Destination": {
        "AlignAVStart": true,
        "ChapterList": [
            {
                "Duration": {
                    "Hours": 0,
                    "Minutes": 0,
                    "Seconds": 9,
                    "Ticks": 819819
                },
                "Name": ""
            }
        ],
        "ChapterMarkers": false,
        "File": "/output/sample-small-2.mp4",
        "InlineParameterSets": false,
        "Mp4Options": {
            "IpodAtom": false,
            "Mp4Optimize": false
        },
        "Mux": "m4v"
    },
    "Filters": {
        "FilterList": [
            {
                "ID": 6,
                "Settings": {
                    "mode": 1
                }
            },
            {
                "ID": 13,
                "Settings": {
                    "crop-bottom": 0,
                    "crop-left": 0,
                    "crop-right": 0,
                    "crop-top": 0,
                    "height": 1520,
                    "width": 2704
                }
            }
        ]
    },
    "Metadata": {
        "Name": "sample"
    },
    "PAR": {
        "Den": 1,
        "Num": 1
    },
    "SequenceID": 0,
    "Source": {
        "Angle": 0,
        "Path": "/output/sample.MP4",
        "Range": {
            "End": 1,
            "Start": 1,
            "Type": "chapter"
        },
        "Title": 1
    },
    "Subtitle": {
        "Search": {
            "Burn": true,
            "Default": false,
            "Enable": false,
            "Forced": false
        },
        "SubtitleList": []
    },
    "Video": {
        "ColorFormat": 0,
        "ColorMatrix": 1,
        "ColorPrimaries": 1,
        "ColorRange": 2,
        "ColorTransfer": 1,
        "Encoder": "qsv_h264",
        "Level": "auto",
        "Options": "",
        "Preset": "quality",
        "Profile": "auto",
        "QSV": {
            "AdapterIndex": 0,
            "AsyncDepth": 0,
            "Decode": false
        },
        "Quality": 20.0,
        "Tune": "",
        "Turbo": false,
        "TwoPass": false
    }
}
[21:01:26] CPU: Intel(R) Celeron(R) J4105 CPU @ 1.50GHz
[21:01:26]  - logical processor count: 4
[21:01:26] Intel Quick Sync Video support: yes
[21:01:26]  - Intel Media SDK hardware: API 1.35 (minimum: 1.3)
[21:01:26]  - Decode support:  h264 hevc (8bit: yes, 10bit: yes) av1 (8bit: yes, 10bit: yes)
[21:01:26]  - H.264 encoder: yes
[21:01:26]     - preferred implementation: hardware (any) via ANY
[21:01:26]     - capabilities (hardware):  breftype icq+la+i+downs vsinfo opt1 opt2+mbbrc+extbrc+trellis+ib_adapt+nmpslice
[21:01:26]  - H.265 encoder: yes (8bit: yes, 10bit: yes)
[21:01:26]     - preferred implementation: hardware (any) via ANY
[21:01:26]     - capabilities (hardware):  lowpower bpyramid icq vsinfo opt1
[21:01:26] hb_scan: path=/output/sample.MP4, title_index=1
udfread ERROR: ECMA 167 Volume Recognition failed
disc.c:333: failed opening UDF image /output/sample.MP4
disc.c:437: error opening file BDMV/index.bdmv
disc.c:437: error opening file BDMV/BACKUP/index.bdmv
bluray.c:2646: nav_get_title_list(/output/sample.MP4/) failed
[21:01:30] bd: not a bd - trying as a stream/file instead
libdvdread: DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
libdvdnav: vm: vm: failed to read VIDEO_TS.IFO
[21:01:30] dvd: not a dvd - trying as a stream/file instead
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/output/sample.MP4':
  Metadata:
    major_brand     : mp41
    minor_version   : 538120216
    compatible_brands: mp41
    creation_time   : 2021-07-21T13:35:52.000000Z
    firmware        : HD8.01.02.50.00
  Duration: 00:00:09.11, start: 0.000000, bitrate: 45371 kb/s
  Stream #0:0(eng): Video: h264 (High) (avc1 / 0x31637661), yuvj420p(pc, bt709), 2704x1520 [SAR 1:1 DAR 169:95], 45082 kb/s, 29.97 fps, 29.97 tbr, 90k tbn, 59.94 tbc (default)
    Metadata:
      creation_time   : 2021-07-21T13:35:52.000000Z
      handler_name    : GoPro AVC  
      vendor_id       : [0][0][0][0]
      encoder         : GoPro AVC encoder
      timecode        : 00:13:33:24
  Stream #0:1(eng): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 188 kb/s (default)
    Metadata:
      creation_time   : 2021-07-21T13:35:52.000000Z
      handler_name    : GoPro AAC  
      vendor_id       : [0][0][0][0]
      timecode        : 00:13:33:24
  Stream #0:2(eng): Data: none (tmcd / 0x64636D74), 0 kb/s (default)
    Metadata:
      creation_time   : 2021-07-21T13:35:52.000000Z
      handler_name    : GoPro TCD  
      timecode        : 00:13:33:24
  Stream #0:3(eng): Data: bin_data (gpmd / 0x646D7067), 54 kb/s (default)
    Metadata:
      creation_time   : 2021-07-21T13:35:52.000000Z
      handler_name    : GoPro MET  
  Stream #0:4(eng): Data: none (fdsc / 0x63736466), 10 kb/s (default)
    Metadata:
      creation_time   : 2021-07-21T13:35:52.000000Z
      handler_name    : GoPro SOS  
[21:01:31] scan: decoding previews for title 1
[21:01:31] scan: audio 0x1: aac, rate=48000Hz, bitrate=188459 English (AAC LC) (2.0 ch) (188 kbps)
[21:01:44] scan: 10 previews, 2704x1520, 29.970 fps, autocrop = 0/0/0/0, aspect 16:9, PAR 1:1, color profile: 1-1-1
[21:01:44] scan: supported video decoders: avcodec qsv
[21:01:44] libhb: scan thread found 1 valid title(s)
[21:01:44] Starting Task: Encoding Pass
[21:01:44] Skipping crop/scale filter
[21:01:44] job configuration:
[21:01:44]  * source
[21:01:44]    + /output/sample.MP4
[21:01:44]    + title 1, chapter(s) 1 to 1
[21:01:44]    + container: mov,mp4,m4a,3gp,3g2,mj2
[21:01:44]    + data rate: 45371 kbps
[21:01:44]  * destination
[21:01:44]    + /output/sample-small-2.mp4
[21:01:44]    + container: MPEG-4 (libavformat)
[21:01:44]      + align initial A/V stream timestamps
[21:01:44]  * video track
[21:01:44]    + decoder: h264 8-bit
[21:01:44]      + bitrate 45082 kbps
[21:01:44]    + filter
[21:01:44]      + Framerate Shaper (mode=1)
[21:01:44]        + frame rate: 29.970 fps -> constant 29.970 fps
[21:01:44]    + Output geometry
[21:01:44]      + storage dimensions: 2704 x 1520
[21:01:44]      + pixel aspect ratio: 1 : 1
[21:01:44]      + display dimensions: 2704 x 1520
[21:01:44]    + encoder: H.264 (Intel Media SDK)
[21:01:44]      + preset:  quality
[21:01:44]      + profile: auto
[21:01:44]      + level:   auto
[21:01:44]      + quality: 20.00 (ICQ)
[21:01:44]      + color profile: 1-1-1
[21:01:44]  * audio track 1
[21:01:44]    + decoder: English (AAC LC) (2.0 ch) (188 kbps) (track 1, id 0x1)
[21:01:44]      + bitrate: 188 kbps, samplerate: 48000 Hz
[21:01:44]    + mixdown: Stereo
[21:01:44]    + encoder: AAC (libfdk_aac)
[21:01:44]      + bitrate: 160 kbps, samplerate: 48000 Hz
[21:01:44] sync: expecting 273 video frames
[21:01:44] hb_display_init: using VA driver 'iHD'
libva info: VA-API version 1.12.0
libva info: User environment variable requested driver 'iHD'
libva info: Trying to open /opt/intel/mediasdk/lib64/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_12
libva info: va_openDriver() returns 0
[21:01:44] encqsvInit: using encode-only path
[21:01:44] encqsvInit: H.264/AVC High profile @ level 5.0
[21:01:44] encqsvInit: TargetUsage 4 AsyncDepth 6
[21:01:44] encqsvInit: GopRefDist 3 GopPicSize 30 NumRefFrame 2
[21:01:44] encqsvInit: BFramesMax 2 BRefType off
[21:01:44] Unknown Header.BufferId=1347637059
[21:01:44] encqsvInit: RateControlMethod LA_ICQ ICQQuality 20 LookAheadDepth 40
[21:01:44] encqsvInit: MBBRC on
[21:01:45] sync: first pts video is 0
[21:01:45] sync: Chapter 1 at frame 1 time 0
[21:01:46] sync: first pts audio 0x1 is 0
[21:02:04] qsv_enc_init: using 'hardware (1) via VAAPI' implementation, API: 1.35
[21:02:26] reader: done. 1 scr changes
[21:02:31] work: average encoding speed for job is 5.689593 fps
[21:02:31] vfr: 273 frames output, 0 dropped and 0 duped for CFR/PFR
[21:02:31] vfr: lost time: 0 (0 frames)
[21:02:31] vfr: gained time: 0 (0 frames) (0 not accounted for)
[21:02:31] aac-decoder done: 426 frames, 0 decoder errors
[21:02:31] h264-decoder done: 273 frames, 0 decoder errors
[21:02:31] sync: got 273 frames, 273 expected
[21:02:31] sync: framerate min 29.970 fps, max 29.970 fps, avg 29.970 fps
[21:02:31] mux: track 0, 273 frames, 9664122 bytes, 8456.47 kbps, fifo 512
[21:02:31] mux: track 1, 428 frames, 182971 bytes, 160.11 kbps, fifo 512
[21:02:31] Finished work at: Sun Jul 25 21:02:31 2021

[21:02:31] libhb: work result = 0
```

Activity log on Windows machine:
```
HandBrake 1.4.0 (2021071800)
OS: Microsoft Windows NT 10.0.18363.0
CPU: Intel(R) Core(TM) i7-6500U CPU @ 2.50GHz
Ram: 16264 MB, 
GPU Information:
  Intel(R) HD Graphics 520 - 27.20.100.8854
  Citrix Indirect Display Adapter - 12.40.44.247
Screen: 1920x1080
Temp Dir: C:\Users\[sensitive]\AppData\Local\Temp\
Install Dir: C:\Program Files\HandBrake\
Data Dir: C:\Users\[sensitive]\AppData\Roaming\HandBrake

-------------------------------------------


# Starting Encode ...
[23:13:50] base preset: GoPro Size Reduce (Modified)
[23:13:50] hb_init: starting libhb thread
[23:13:50] Starting work at: Fri Jul 23 23:13:50 2021
[23:13:50] 1 job(s) to process
[23:13:50] json job:
{
  "Audio": {
    "AudioList": [
      {
        "Bitrate": 320,
        "DRC": 0,
        "Encoder": "av_aac",
        "Gain": 0,
        "Mixdown": 4,
        "NormalizeMixLevel": false,
        "Samplerate": 0,
        "Track": 0,
        "DitherMethod": 0
      }
    ],
    "CopyMask": [
      "copy:aac",
      "copy:ac3",
      "copy:dtshd",
      "copy:dts",
      "copy:eac3",
      "copy:flac",
      "copy:mp3",
      "copy:truehd",
      "copy:mp2"
    ],
    "FallbackEncoder": "ac3"
  },
  "Destination": {
    "ChapterList": [
      {
        "Name": "\u0413\u043B\u0430\u0432\u0430 1"
      }
    ],
    "ChapterMarkers": true,
    "AlignAVStart": false,
    "File": "C:\\Users\\[sensitive]\\Downloads\\sample-convert-small-2.mp4",
    "Mp4Options": {
      "IpodAtom": false,
      "Mp4Optimize": false
    },
    "Mux": "av_mp4"
  },
  "Filters": {
    "FilterList": [
      {
        "ID": 13,
        "Settings": {
          "crop-bottom": "0",
          "crop-left": "0",
          "crop-right": "0",
          "crop-top": "0",
          "height": "1520",
          "width": "2704"
        }
      },
      {
        "ID": 6,
        "Settings": {
          "mode": "1"
        }
      }
    ]
  },
  "PAR": {
    "Num": 1,
    "Den": 1
  },
  "Metadata": {},
  "SequenceID": 0,
  "Source": {
    "Angle": 1,
    "Range": {
      "Type": "chapter",
      "Start": 1,
      "End": 1
    },
    "Title": 1,
    "Path": "D:\\[sensitive]\\GH019723.MP4"
  },
  "Subtitle": {
    "Search": {
      "Burn": false,
      "Default": false,
      "Enable": false,
      "Forced": false
    },
    "SubtitleList": []
  },
  "Video": {
    "Encoder": "qsv_h264",
    "Level": "auto",
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "",
    "Preset": "quality",
    "Profile": "auto",
    "Quality": 20,
    "QSV": {
      "Decode": true
    }
  }
}
[23:13:50] CPU: Intel(R) Core(TM) i7-6500U CPU @ 2.50GHz
[23:13:50]  - Intel microarchitecture Skylake
[23:13:50]  - logical processor count: 4
[23:13:50] Intel Quick Sync Video support: yes, gpu list: 0
[23:13:50] Intel Quick Sync Video integrated adapter with index 0
[23:13:50]  - Intel Media SDK hardware: API 1.33 (minimum: 1.3)
[23:13:50]  - Decode support:  h264 hevc (8bit: yes, 10bit: no)
[23:13:50]  - H.264 encoder: yes
[23:13:50]     - preferred implementation: hardware (1) via D3D11
[23:13:50]     - capabilities (hardware):  breftype icq+la+i+downs vsinfo opt1 opt2+mbbrc+extbrc+trellis+ib_adapt+nmpslice
[23:13:50]  - H.265 encoder: yes (8bit: yes, 10bit: no)
[23:13:50]     - preferred implementation: hardware (1) via D3D11
[23:13:50]     - capabilities (hardware):  bpyramid icq vsinfo opt1
[23:13:50] hb_scan: path=D:\GoPro\[sensitive]\GH019723.MP4, title_index=1
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:333: failed opening UDF image D:\GoPro\[sensitive]\GH019723.MP4
src/libbluray/disc/disc.c:437: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:437: error opening file BDMV\BACKUP\index.bdmv
src/libbluray/bluray.c:2646: nav_get_title_list(D:\[sensitive]\GH019723.MP4\) failed
[23:13:50] bd: not a bd - trying as a stream/file instead
libdvdread: DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
libdvdnav: vm: vm: failed to read VIDEO_TS.IFO
[23:13:50] dvd: not a dvd - trying as a stream/file instead
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'D:\[sensitive]\GH019723.MP4':
  Metadata:
    major_brand     : mp41
    minor_version   : 538120216
    compatible_brands: mp41
    creation_time   : 2021-07-21T13:35:52.000000Z
    firmware        : HD8.01.02.50.00
  Duration: 00:00:09.11, start: 0.000000, bitrate: 45371 kb/s
  Stream #0:0(eng): Video: h264 (High) (avc1 / 0x31637661), yuvj420p(pc, bt709), 2704x1520 [SAR 1:1 DAR 169:95], 45082 kb/s, 29.97 fps, 29.97 tbr, 90k tbn, 59.94 tbc (default)
    Metadata:
      creation_time   : 2021-07-21T13:35:52.000000Z
      handler_name    : GoPro AVC
      vendor_id       : [0][0][0][0]
      encoder         : GoPro AVC encoder
      timecode        : 00:13:33:24
  Stream #0:1(eng): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 188 kb/s (default)
    Metadata:
      creation_time   : 2021-07-21T13:35:52.000000Z
      handler_name    : GoPro AAC
      vendor_id       : [0][0][0][0]
      timecode        : 00:13:33:24
  Stream #0:2(eng): Data: none (tmcd / 0x64636D74), 0 kb/s (default)
    Metadata:
      creation_time   : 2021-07-21T13:35:52.000000Z
      handler_name    : GoPro TCD
      timecode        : 00:13:33:24
  Stream #0:3(eng): Data: bin_data (gpmd / 0x646D7067), 54 kb/s (default)
    Metadata:
      creation_time   : 2021-07-21T13:35:52.000000Z
      handler_name    : GoPro MET
  Stream #0:4(eng): Data: none (fdsc / 0x63736466), 10 kb/s (default)
    Metadata:
      creation_time   : 2021-07-21T13:35:52.000000Z
      handler_name    : GoPro SOS
[23:13:50] scan: decoding previews for title 1
[23:13:50] scan: audio 0x1: aac, rate=48000Hz, bitrate=188459 English (AAC LC) (2.0 ch) (188 kbps)
[23:13:52] scan: 10 previews, 2704x1520, 29.970 fps, autocrop = 0/0/0/0, aspect 16:9, PAR 1:1, color profile: 1-1-1
[23:13:52] scan: supported video decoders: avcodec qsv
[23:13:52] libhb: scan thread found 1 valid title(s)
[23:13:52] Starting Task: Encoding Pass
[23:13:52] Skipping crop/scale filter
[23:13:52] work: track 1, dithering not supported by codec
[23:13:52] work: only 1 chapter, disabling chapter markers
[23:13:52] job configuration:
[23:13:52]  * source
[23:13:52]    + D:\[sensitive]\GH019723.MP4
[23:13:52]    + title 1, chapter(s) 1 to 1
[23:13:52]    + container: mov,mp4,m4a,3gp,3g2,mj2
[23:13:52]    + data rate: 45371 kbps
[23:13:52]  * destination
[23:13:52]    + C:\Users\[sensitive]\Downloads\sample-convert-small-2.mp4
[23:13:52]    + container: MPEG-4 (libavformat)
[23:13:52]  * video track
[23:13:52]    + decoder: h264_qsv 8-bit
[23:13:52]      + bitrate 45082 kbps
[23:13:52]    + filter
[23:13:52]      + Framerate Shaper (mode=1)
[23:13:52]        + frame rate: 29.970 fps -> constant 29.970 fps
[23:13:52]    + Output geometry
[23:13:52]      + storage dimensions: 2704 x 1520
[23:13:52]      + pixel aspect ratio: 1 : 1
[23:13:52]      + display dimensions: 2704 x 1520
[23:13:52]    + encoder: H.264 (Intel Media SDK)
[23:13:52]      + preset:  quality
[23:13:52]      + profile: auto
[23:13:52]      + level:   auto
[23:13:52]      + quality: 20.00 (ICQ)
[23:13:52]      + color profile: 1-1-1
[23:13:52]  * audio track 1
[23:13:52]    + decoder: English (AAC LC) (2.0 ch) (188 kbps) (track 1, id 0x1)
[23:13:52]      + bitrate: 188 kbps, samplerate: 48000 Hz
[23:13:52]    + mixdown: Stereo
[23:13:52]    + encoder: AAC (libavcodec)
[23:13:52]      + bitrate: 320 kbps, samplerate: 48000 Hz
[23:13:52] sync: expecting 273 video frames
[23:13:52] qsv: integrated qsv adapter with index 0 has been selected
[23:13:53] encqsvInit: using encode-only path
[23:13:53] encqsvInit: H.264/AVC High profile @ level 5.0
[23:13:53] encqsvInit: TargetUsage 4 AsyncDepth 4
[23:13:53] encqsvInit: GopRefDist 3 GopPicSize 30 NumRefFrame 2
[23:13:53] encqsvInit: BFramesMax 2 BRefType off
[23:13:53] Unknown Header.BufferId=1347637059
[23:13:53] encqsvInit: RateControlMethod LA_ICQ ICQQuality 20 LookAheadDepth 40
[23:13:54] sync: first pts video is 0
[23:13:54] sync: "Глава 1" (1) at frame 1 time 0
[23:13:54] sync: first pts audio 0x1 is 0
[23:13:55] qsv_enc_init: using 'hardware (1) via D3D11' implementation, API: 1.33
[23:14:05] reader: done. 1 scr changes
[23:14:11] work: average encoding speed for job is 19.586323 fps
[23:14:11] vfr: 273 frames output, 0 dropped and 0 duped for CFR/PFR
[23:14:11] vfr: lost time: 0 (0 frames)
[23:14:11] vfr: gained time: 0 (0 frames) (0 not accounted for)
[23:14:11] aac-decoder done: 426 frames, 0 decoder errors
[23:14:11] h264_qsv-decoder done: 273 frames, 0 decoder errors
[23:14:11] sync: got 273 frames, 273 expected
[23:14:11] sync: framerate min 29.970 fps, max 29.970 fps, avg 29.970 fps
[23:14:11] mux: track 0, 273 frames, 11645354 bytes, 10190.12 kbps, fifo 256
[23:14:11] mux: track 1, 427 frames, 268430 bytes, 234.89 kbps, fifo 512
[23:14:11] Finished work at: Fri Jul 23 23:14:11 2021
[23:14:11] libhb: work result = 0

# Job Completed!

```
```


### What Operating System are you running?

Windows 10, docker, Ubuntu 20.04

### What version of HandBrake are you running?

1.4.0, 1.3.1

### Where did you download HandBrake from?

dockerhub, handbrake.fr

--- a/libhb/qsv_common.c
+++ b/libhb/qsv_common.c
@@ -1842,6 +1842,12 @@ int hb_qsv_full_path_is_enabled(hb_job_t *job)
         return 0;
     }
 
+    // scale_qsv filter can't convert from full to limited range
+    if (job->title->color_range == AVCOL_RANGE_JPEG && job->color_range == AVCOL_RANGE_MPEG)
+    {
+        return 0;
+    }
+
     qsv_full_path_is_enabled = (hb_qsv_decode_is_enabled(job) &&
         info && hb_qsv_implementation_is_hardware(info->implementation) &&
         device_check_succeeded && job->qsv.ctx && !job->qsv.ctx->num_cpu_filters);

